dbms/src/Core/SettingsCommon.cpp

```diff
@@ -552,9 +546,10 @@ void SettingChar::set(char x)
 
 void SettingChar::set(const String & x)
 {
-    checkStringIsACharacter(x);
-    value = x[0];
-    changed = true;
+    if (x.size() > 1)
+        throw Exception("A setting's value string has to be an exactly one character long", ErrorCodes::SIZE_OF_FIXED_STRING_DOESNT_MATCH);
+    char c = (x.size() == 1) ? x[0] : '\0';
```

Comment:
How can we now distinguish the empty setting and 'zero'-char setting values?

Reply:
We don't distinguish them. The following two declaration mean the same:

```
create table simple (
  t UInt64,
  url String
) ENGINE=Kafka SETTINGS kafka_broker_list = 'localhost:9092',
                        kafka_topic_list = 'topic',
                        kafka_group_name = 'mygroup',
                        kafka_format = 'Protobuf',
                        kafka_row_delimiter = '',
                        kafka_schema = 'simple:AccessLog';
```
and
```
create table simple (
  t UInt64,
  url String
) ENGINE=Kafka SETTINGS kafka_broker_list = 'localhost:9092',
                        kafka_topic_list = 'topic',
                        kafka_group_name = 'mygroup',
                        kafka_format = 'Protobuf',
                        kafka_row_delimiter = '\0',
                        kafka_schema = 'simple:AccessLog';
```